solutions/platinum/hr-SubtreesPaths.mdx

```diff
@@ -0,0 +1,196 @@
+---
+id: hr-SubtreesPaths
+source: HR
+title: Subtrees & Paths
+author: Rameez Parwez
+---
+
+Given a rooted tree of $N$ nodes, where each node is uniquely numbered in between $[1...N]$. The node $1$ is the root of the tree. Each node has an integer value which is initially $0$.
+
+You need to perform the following two kinds of queries on the tree:
+- $add$ $t$ $value$: Add $value$ to all nodes in subtree rooted at $t$
+- $max$ $a$ $b$: Report maximum value on the path from $a$ to $b$
+
+## Intuition
+
+From the question statement, it is clear that this is a Heavy-Light Decomposition(HLD). To handle the query operations, we will combine HLD with Lazy Segment Tree.
```

Comment:
? not clear

The problem asks us to find X, so we can use HLD.

Reply:
implemented